testplans/bitswap/main.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/testground/sdk-go/network"
+	"github.com/testground/sdk-go/run"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+
+	bitswap "github.com/ipfs/go-bitswap"
+	bsnet "github.com/ipfs/go-bitswap/network"
+	block "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	datastore "github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	exchange "github.com/ipfs/go-ipfs-exchange-interface"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multihash"
+)
+
+var (
+	testcases = map[string]interface{}{
+		"speed-test": run.InitializedTestCaseFn(runSpeedTest),
+	}
+	networkState  = sync.State("network-configured")
+	readyState    = sync.State("ready-to-publish")
+	readyDLState  = sync.State("ready-to-download")
+	doneState     = sync.State("done")
+	providerTopic = sync.NewTopic("provider", &peer.AddrInfo{})
+	blockTopic    = sync.NewTopic("blocks", &multihash.Multihash{})
+)
+
+func main() {
+	run.InvokeMap(testcases)
+}
+
+func runSpeedTest(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
+	runenv.RecordMessage("running speed-test")
+	ctx := context.Background()
+	initCtx.NetClient.MustConfigureNetwork(ctx, &network.Config{
+		Network:        "default",
+		Enable:         true,
+		Default:        network.LinkShape{},
```

Comment:
For regression tests (and code-based accountability), link parameters should be explicit. The reason: If someone changes the LinkShape defaults in the future (in the sdk repo), there will be no code trace (in this test plan) that indicates what happened and people will wonder why the regression results are different.